File: 69-Sqrt(x)/solution.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x < 2:
            return x

        start, end = 1, x / 2

        while start <= end:
            mid = start + (end - start) / 2
            if x / mid < mid:
                end = mid - 1
            else:
                start = mid + 1

        return end
        

File: 69-Sqrt(x)/solution.c
int mySqrt(int x) {
    if (x < 2)
        return x;
    int left = 1, right = x / 2;
    int mid;
    int last_mid;
    
    while (left <= right)
    {
        mid = left + (right - left) / 2;
        if (x / mid > mid)
        {
            left = mid + 1;
            last_mid = mid;
        }
        else if (x / mid < mid)
        {
            right = mid - 1;
        }
        else
            return mid;
    }
    return last_mid;
}